src/averageAges.js

```diff
@@ -37,9 +45,23 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let womens = people.filter(person => person.sex === 'f');
+
+  if (withChildren === true) {
+    womens = womens.filter(woman => people.some(person =>
+      person.mother === woman.name));
+  }
+
+  const totalAge = womens.reduce((sum, person) => sum
+    + (person.died - person.born), 0);
+
+  const averageAge = parseFloat((totalAge / womens.length).toFixed(2));
```

Comment:
Using 'parseFloat' to round to two decimal places is a good practice, but it's used twice in almost identical contexts. Consider creating a helper function for this to avoid code duplication.